sunpy/instr/aia.py

```diff
@@ -49,7 +50,15 @@ def aiaprep(aiamap):
         scale = 0.6*u.arcsec # pragma: no cover # can't test this because it needs a full res image
     scale_factor = aiamap.scale.x / scale
 
-    newmap = aiamap.rotate(recenter=True, scale=scale_factor.value, missing=aiamap.min())
+    tempmap = aiamap.rotate(recenter=True, scale=scale_factor.value, missing=aiamap.min())
+
+    # extract center from padded aiamap.rotate output
+    # crpix1 and crpix2 will be equal (recenter=True), as aiaprep does not work with submaps
```

Comment:
How do you assure it doesn't work for a submap? I've just tried and it didn't show any error or warning that I should not being doing this. Maybe raising an error if the dimensions are not right would be better than just proceed as normal and produce bad data:

```python
>>> saia_map = aia_map.submap(u.Quantity([30, 809], u.pix), u.Quantity([35, 789], u.pix))
>>> saia_map_prep = sunpy.instr.aia.aiaprep(saia_map)
>>> saia_map.dimensions
Pair(x=<Quantity 779.0 pix>, y=<Quantity 754.0 pix>)
>>> saia_map_prep.dimensions
Pair(x=<Quantity 754.0 pix>, y=<Quantity 741.0 pix>)
>>> saia_map.xrange     
>>> <Quantity [-1156.8,  712.8] arcsec>
>>> saia_map_prep.xrange    
<Quantity [-903.6, 906. ] arcsec>
>>> saia_map.yrange     
<Quantity [-1144.8,  664.8] arcsec>
>>> saia_map_prep.yrange
<Quantity [-873.6, 904.8] arcsec>
```